ext/liquid_c/tokenizer.c

```diff
@@ -66,20 +67,29 @@ void tokenizer_next(tokenizer_t *tokenizer, token_t *token)
 
     token->str = cursor;
     token->type = TOKEN_RAW;
+    token->lstrip = 0;
+    token->rstrip = 0;
 
     while (cursor < last) {
         if (*cursor++ != '{')
             continue;
 
         char c = *cursor++;
+        if (cursor <= last && *cursor == '-') {
+          cursor++;
+          token->rstrip = 1;
```

Comment:
If the following `if (c != '%' && c != '{')` branch is taken, then this change won't be reversed.